File: qa-screenshots.mjs
import { chromium } from 'playwright';
import { mkdirSync } from 'fs';

const BASE = 'http://localhost:3000';
const OUT  = 'C:/Users/ofekd/CampingStore/qa-shots';
mkdirSync(OUT, { recursive: true });

const PAGES = [
  { path: '/',                         name: 'home' },
  { path: '/shop',                     name: 'shop' },
  { path: '/shop/camp-kitchen',        name: 'shop-camp-kitchen' },
  { path: '/shop/vehicle-gear',        name: 'shop-vehicle-gear' },
  { path: '/shop/sleeping',            name: 'shop-sleeping' },
  { path: '/product/espresso-nayad',   name: 'product-espresso' },
  { path: '/product/plasma-lighter',   name: 'product-lighter' },
  { path: '/product/medeflatorm',      name: 'product-deflators' },
  { path: '/product/arasal-shtat',     name: 'product-hammock' },
  { path: '/cart',                     name: 'cart' },
  { path: '/checkout',                 name: 'checkout' },
];

const VIEWPORTS = [
  { width: 1440, height: 900,  label: 'desktop' },
  { width: 390,  height: 844,  label: 'mobile'  },
];

const browser = await chromium.launch();

for (const vp of VIEWPORTS) {
  const ctx  = await browser.newContext({ viewport: { width: vp.width, height: vp.height } });
  const page = await ctx.newPage();

  for (const pg of PAGES) {
    await page.goto(`${BASE}${pg.path}`, { waitUntil: 'networkidle', timeout: 15000 });
    await page.waitForTimeout(600);
    const file = `${OUT}/${vp.label}-${pg.name}.png`;
    await page.screenshot({ path: file, fullPage: true });
    console.log(`✓ ${vp.label} ${pg.path}`);
  }

  await ctx.close();
}

await browser.close();
console.log('\nAll screenshots saved to', OUT);

File: qa-screenshots.py
from playwright.sync_api import sync_playwright
import os

BASE = "http://localhost:3000"
OUT  = r"C:\Users\ofekd\CampingStore\qa-shots"
os.makedirs(OUT, exist_ok=True)

PAGES = [
    ("/",                       "home"),
    ("/shop",                   "shop"),
    ("/shop/camp-kitchen",      "shop-camp-kitchen"),
    ("/shop/vehicle-gear",      "shop-vehicle-gear"),
    ("/product/espresso-nayad", "product-espresso"),
    ("/product/plasma-lighter", "product-lighter"),
    ("/product/medeflatorm",    "product-deflators"),
    ("/product/arasal-shtat",   "product-hammock"),
    ("/cart",                   "cart"),
    ("/checkout",               "checkout"),
]

VIEWPORTS = [
    (1440, 900,  "desktop"),
    (390,  844,  "mobile"),
]

with sync_playwright() as p:
    browser = p.chromium.launch()
    for w, h, label in VIEWPORTS:
        ctx  = browser.new_context(viewport={"width": w, "height": h})
        page = ctx.new_page()
        for path, name in PAGES:
            try:
                page.goto(f"{BASE}{path}", wait_until="networkidle", timeout=15000)
                page.wait_for_timeout(800)
                fp = os.path.join(OUT, f"{label}-{name}.png")
                page.screenshot(path=fp, full_page=True)
                print(f"OK {label}  {path}")
            except Exception as e:
                print(f"ERR {label}  {path}  ERROR: {e}")
        ctx.close()
    browser.close()

print("Done. Saved to " + OUT)
